frontend/src/features/search/Search.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useEffect, useState, useRef } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { useHistory } from 'react-router-dom'
+
+import { loadSearchResults } from './state/searchActions'
+import { selectSearchResults } from './state/searchState'
+import NavBar from '../navbar/NavBar';
+import BigSearchBox from './BigSearchBox';
+import SearchResultItem from './SearchResultItem'
+import PageControl, { PageChangeObject } from '../../chrome/PageControl'
+import Footer from '../footer/Footer'
+
+import { QueryParams } from '../../qri/queryParams'
+
+interface SearchProps {
+  queryParams: QueryParams
+}
+
+const Search: React.FC<SearchProps> = ({ queryParams }) => {
+  const dispatch = useDispatch()
+  const history = useHistory()
+  const searchResults = useSelector(selectSearchResults)
+
+  // keep track of the current query and page in local state
+  // TODO(chriswhong): add sorting
+  const [q, setQ] = useState(queryParams.q)
+  const [page, setPage] = useState(queryParams.page)
+
+  const prevQ = useRef()
+
+  // TODO(chriswhong): get pageInfo from state, this is a placeholder to
+  // lay out and test PageControl
+  const pageInfo = {
+    resultCount: 68,
+    page,
+    pageSize: 25,
+    pageCount: 3
+  }
```

Comment:
Are we still not getting valid values back from server responses? We should get this addressed before merging

Reply:
got it working now